### PR TITLE
[MIRROR] [reviewpls] Adds moodlets to the game - [Please give suggestions for trait additions in comments]

### PR DESCRIPTION
**Pull request by ghost**
Original PR: https://github.com/tgstation/tgstation/pull/35475
--------------------
:cl: Floyd / Qustinnus (Sprites by Ausops, Some moodlets by Ike709)
add: Adds mood, which can be found by clicking on the face icon on your screen.
add: Adds various moodlets which affect your mood. Try eating your favourite food, playing an arcade game, reading a book, or petting a doggo to increase your moo. Also be sure to take care of your hunger on a regular basis, like always.
add: Adds config option to disable/enable mood.
add: Indoor area's now have a beauty var defined by the amount of cleanables in them, (We can later expand this to something like rimworld, where structures could make rooms more beautiful). These also affect mood. (Janitor now has gameplay purpose besides slipping and removing useless decals)
remove: Removes hunger slowdown, replacing it with slowdown by being depressed
imageadd: Icons for mood states and depression states
/:cl:

# What this PR is
This PR adds a system that allows player to gain and lose moodlets based on events occuring to, and around them. These events then give the player a mood value based on what it is. For example a hug could give you +1 mood, while being stabbed in the eye with a screwdriver can give -5 mood. All these moodlets together determine the mood of your character which currently affects the following things:

1. Movement speed - If you are very sad you move slower. Replacing movement slow from hunger. (hunger now instead affects mood)
2. Screen blur - If you are sad you gain an overlay that slightly blurs the screen, increasing in severity as you get sadder.
3. Interaction / do after speed - If you are sad or happy your interaction speed with things such as handcuffs is changed. with a 25% longer time if you are sad, or 10% shorter time if you are extremely happy.
4. Hunger rate - You gain hunger slower if you are very happy.

I'd love more suggestions for way to make gaining good moods appealing in a way that doesn't affect combat too much, so please post them! I would also love if people tell me about other cool possibilities for things that give moodlets, especially things that give positive moodlets are lacking right now.

image
_A person in a fairly good mood. (third best happy mood)_

image
_A person in an almost fully suicidal mood (second worst)_

Also some of the various moodlets I have already added in this PR, with tens of others being able to be added after or during this PR. (timeout means time in milliseconds before the event dissapears on it own, if it does)
```
/datum/mood_event/creampie
   description = "<span class='warning'>I've been creamed. Tastes like pie flavor.</span>\n"
   mood = -2
   timeout = 1800

/datum/mood_event/delam //SM delamination
   description = "<span class='boldwarning'>Those God damn engineers can't do anything right...</span>\n"
   mood = -6
   timeout = 3000

/datum/mood_event/shameful_suicide //suicide_acts that return SHAME, like sord
  description = "<span class='boldwarning'>I can't even end it all!</span>\n"
  mood = -10
  timeout = 600

/datum/mood_event/blessing
   description = "<span class='green'>I've been blessed.</span>\n"
   mood = 3
   timeout = 3000

/datum/mood_event/honk
   description = "<span class='green'>Maybe clowns aren't so bad after all. Honk!</span>\n"
   mood = 2
   timeout = 2400

/datum/mood_event/broken_vow //Used for when mimes break their vow of silence
  description = "<span class='boldwarning'>I have brought shame upon my name, and betrayed my fellow mimes by breaking our sacred vow...</span>\n"
  mood = -5

/datum/mood_event/drugs/withdrawal_severe
   mood = -7

/datum/mood_event/drugs/withdrawal_severe/add_effects(name)
   description = "<span class='boldwarning'>Oh god I need some [name]</span>\n"
```

# Why
I think the game should not enforce every piece of gameplay with rules such as some servers do, I think we can instead create cool interaction between players by rewarding them in a minimal way for it, which is why I created this system. This system allows players to gain positive moodlets for doing generally positive things. Reinforcing them to seek out certain interactions that would otherwise not necesarily be useful. This partially makes roleplaying a bit more appealing, while not forcing it.

A primary concern for this is definitely that some players might not enjoy being forced to do tasks they might dislike like showering etc. Which is why I ensured the following in this PR
1. The effects of not engaging with moodlets are not negative, unless you are severely abusive to yourself at which point it will affect you about the same as hunger normally would.
2. It is extremely difficult to get severely sad without intent.
3. I make it so the benefits of gaining moodlets are beneficial, but you aren't always losing because someone else always has their mood on perfect.

If the concerns are raised that the effects are still too large, I will try to lower them even more, or even fully remove them, leaving the system to a more passive indication of how "good" you are doing.

A secondary concern I can imagine that might be raised is that this PR is too high RP, with which I disagree, HRP is full of passive rp, where you have to pretend you are doing something like "/me unwraps the burger". Instead TG is all about doing things and actively playing. Which is why this system works fine, as you are rewarded for performing actions.

A final concern I've often heard is that this system tells you how your character feels which might not be preferable for some types of players. "My character doesn't care if X happens to them". For this I'd like to wait on Xhuis' trait system to be merged, and I will create various traits to ensure I can also remove this concern. Adding traits that block certain moodlets or maybe even add new ones (alcoholic trait gets good moodlets from alcohol, for example).

If I missed any information, I might have mentioned it inside of: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=13943

Lastly, if anyone has better sprites for either the face icons or the "depression blur", I'd love to hear from you!

# Special thanks
KMC - Old Sprites for mood faces
Ausops - New sprites for mood faces
Ike709 - Coded several of the positive/negative moodlets

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
/TG/ LITERALLY JUST ADDED ROLEPLAY RULES TO THE SERVER, EXCEPT NOW IT'S ENFORCED VIA GAMEPLAY INSTEAD OF THROUGH RULES
killing self 3

**DENY THIS DENY THIS DENY THIS DENY THIS DENY THIS DENY THIS DENY THIS DENY THIS DENY THIS**

# THE ABSOLUTE STATE

To be more fair, it's a cool feature but it absolutely does not belong on Hippiestation.

Don't accept it, though. Seriously. If you do, disable it immediately in the config - don't sit on your asses like you did with the god awful night lighting system.

**Comment by earthmoonjune** [comment]
delet this

**Comment by GuyonBroadway** [comment]
Keep calm, its config controlled so we can just turn it off and pretend its not there.

**Comment by Manly-Dwarf** [comment]
As I said, night lighting was a config option. Oops it was never disabled and now you can't see shit.

**Comment by GuyonBroadway** [comment]
image

Anything config must be controlled by erik. If you wanna see config shit changed, ping erik. He might say no if you are just one lone autist reeeing into the dark.

**Comment by Manly-Dwarf** [comment]
Yes, it's controlled by Erik but why not merge it when he's online and then ask him to disable it once it's in the game?
If you want to give this a test run, that's okay, but it might need tweaking to be fit for Hippie. Besides, if the test run goes like how night lights are going now (nobody likes them save for like 2 dudes) then I'll beat you up and take your lunch money.

**Comment by GuyonBroadway** [comment]
Cus headcoder Carbon is reeing at us tgo get on these PR's.

also

image

On hippie people who talk about the good are drowned out by Hyperactive-Pepe-stuck-in-the-megaphone-factory levels of autistic screeching about the bad. 

While I think night-lighing could use work and was unnecessary does not mean that I can just discount the feelings of anyone else.

**Comment by Manly-Dwarf** [comment]
Now I hate to sound like a pedant, but [Oprayx hasn't been on for more than two rounds in the past 30 days](https://tools.hippiestation.com/stats/stats/adminConnections.php) unless he never actually administrates the game - and only plays it. I doubt that's enough for night lighting to really sink in and start annoying you, especially when it's every round.

The admins might like it but from what I see in-game, barely anyone does. It might sound like "hyperactive-pepe-stuck-in-the-megaphone-screeching" but I see about two people saying it's good, and most of them just say it's "comfy" or something like that.

It's probably a good idea to leave the shit flinging to the suggestions forum for now, so let's just agree to disagree.

Also if Carbon is reeing at you to merge crappy PRs/config-based PRs, then beat him up and take his ham discs.

**Comment by ghost** [comment]
git gud noobs

**Comment by tortellinitony** [comment]
Merge and then disable via config

**Comment by Pyko1** [comment]
Naah merge, and let it run for a day or 2 and then decide.

**Comment by ghost** [comment]
Low rp doesn't benefit from the game telling you how your character feels. Should be disabled config wise.

**Comment by Pyko1** [comment]
yeah but would be interesting to see couple rounds on hippie with them, just to get the generic idea how it plays on here. since they apparently going to develop it even further.

**Comment by ghost** [comment]
or just disable it and wait until /tg/ develops it and play tests it since when you turn the on switch anyway, you'll inherit all their changes unless we override them beforehand

us opting in now doesn't make development on them happen faster unless we actively choose to modify to suit hippie

**Comment by 2Dollarbill** [comment]
Delt this

**Comment by GuyonBroadway** [comment]
Just throwing an idea out wouldn't it be neat if players could opt to start the game with a "Nanotrasen mood regulator implant" which does nothing besides keep your mood perfectly neutral at all times, no benefits, no negatives. Spare implants would be in medbay storage in case people decide to give the system a go and don't like it.

It'd also be neat if traitors had things that could give people really bad moods, not to mention traitors being immune to moodlet effects due to syndicate training.

**Comment by ghost** [comment]
@GuyonBroadway traitors are already "immune", in the sense they get a moodlet that gives  like 12 mood or something

**Comment by ghost** [comment]
also idk if hippie has traits turned on but I'm definitely ok with PRing a change to the apathetic trait (which currently makes moodies only affect you 80% as much), and make it affect you like 10% or not at all.

feel free to message me any changes you'd like, I can make any seperate changes for hippie.

**Comment by GuyonBroadway** [comment]
I did think that it might be nice if players could get their hands on an implant of something that sets their mood to neutral. No downsides (besides hunger) but no benefits either. 

Also I once had the idea that every so often your character might "fancy" some kind of food or drink to make the bartender/chef have a modicum of use and so that having various food types makes sense. Moodlets would be a good way to impliment that.

Also add happy pills.

**Comment by Manly-Dwarf** [comment]
I really like the concept for this PR, I just think that it could use some tuning on the Hippie-Side to make it better. Otherwise it's just completely incompatible.

It's perfect for Dwarves though. Absolutely incredible.

**Comment by Pyko1** [comment]
I mean its not hippie shit as it is, but it could be.

Mainly have moods not really affect that much on their own, but have party pooper things to do, wizard spells to summon moday to ruin everyones day and etc.

**Comment by JohnMcLaime** [comment]
Inb4 "i killed him cuz he wasnt getting depressed by my food, thus he must be a traitor"

**Comment by ghost** [comment]
No, traitor moodlet is hidden and when examined other players might think youre depresssed even though you arent if you have the traitor moodlet.

**Comment by JohnGinnane** [comment]
we have issue with hud placement

image

**Comment by ghost** [comment]
You need to change the define for hud placement, tg has no stamina like that so

**Comment by JohnGinnane** [comment]
Cool, will look into that, thanks

**Comment by ghost** [comment]
This is dumb, let's just merge it and disable it.

**Comment by JohnGinnane** [comment]
I've moved the icon down so it's below the stamina icon

File: code/controllers/configuration/entries/game_options.dm [hunk 1]
```diff
@@ -50,6 +50,9 @@
 
 /datum/config_entry/flag/allow_ai	// allow ai job
 
+/datum/config_entry/flag/disable_human_mood
+	config_entry_value = TRUE
```
Comment (JohnGinnane) [inline code/controllers/configuration/entries/game_options.dm:54]: @Chnkr not sure if you need to mimic this with our config but this is what you need to do to disable it

**Review by JohnGinnane — APPROVED** [review]

File: code/controllers/configuration/entries/game_options.dm (hunk 1 above)
Comment (ghost) [reply code/controllers/configuration/entries/game_options.dm:54]: kk

**Pull request merged**